src/material.rs

```diff
@@ -1,45 +1,220 @@
+//! Material parameters for mesh rendering.
+
 use render::BasicPipelineState;
 use scene::Color;
 use texture::Texture;
 
-/// Material is the enhancement of Texture that is used to setup appearance of [`Mesh`](struct.Mesh.html).
-#[allow(missing_docs)]
+pub use self::basic::Basic;
+
+/// Basic material parameters.
+pub mod basic {
+    use super::*;
+
+    /// Parameters for a basic mesh material.
+    #[derive(Clone, Debug, Default)]
+    pub struct Basic {
+        /// Solid color applied in the absense of `map`.
+        ///
+        /// Default: `0x000000` (black).
+        pub color: Color,
+
+        /// Texture applied using the mesh texture co-ordiantes.
+        ///
+        /// Default: `None`.
+        pub map: Option<Texture<[f32; 4]>>,
+
+        /// Specifies which pipeline should be used to render the mesh.
+        ///
+        /// Default: `Solid`.
+        pub pipeline: Pipeline,
+    }
+
+    /// Specifies which pipeline should be used to render the mesh.
+    #[derive(Clone, Debug)]
+    pub enum Pipeline {
+        /// Renders the mesh as a solid.
+        Solid,
+
+        /// Renders the mesh as a wireframe.
+        Wireframe,
+
+        /// Renders the mesh with a custom pipeline state object.
+        Custom(BasicPipelineState),
+    }
+
+    impl Default for Pipeline {
+        fn default() -> Self {
+            Pipeline::Solid
+        }
+    }
+}
+
+/// Parameters for a Lamberian diffusion reflection model.
+#[derive(Clone, Debug, Default)]
+pub struct Lambert {
+    /// Solid color applied in the absense of `map`.
+    ///
+    /// Default: `0x000000` (black).
+    pub color: Color,
+
+    /// Specifies whether lighting should be constant over faces.
+    ///
+    /// Default: `false` (lighting is interpolated across faces).
+    pub flat: bool,
+}
+
+/// Parameters for a PBR (physically based rendering) lighting model.
+#[derive(Clone, Debug)]
+pub struct Pbr {
+    /// Solid base color applied in the absense of `base_color_map`.
+    ///
+    /// Default: `0xFFFFFF` (white).
```

Comment:
All other defaults are black. Should we unify, i.e. make all 'base colours' white by default?

Reply:
^^ I decided to make all base colours `WHITE` by default.